example/image-classification/train_cifar10.py

```diff
@@ -36,13 +36,28 @@
 # download data if necessary
 def _download(data_dir):
     if not os.path.isdir(data_dir):
-        os.system("mkdir " + data_dir)
+        os.makedirs(data_dir)  # Recursive directory creation
     os.chdir(data_dir)
     if (not os.path.exists('train.rec')) or \
        (not os.path.exists('test.rec')) :
-        os.system("wget http://webdocs.cs.ualberta.ca/~bx3/data/cifar10.zip")
-        os.system("unzip -u cifar10.zip")
-        os.system("mv cifar/* .; rm -rf cifar; rm cifar10.zip")
+        url = 'http://webdocs.cs.ualberta.ca/~bx3/data/cifar10.zip'
+        if sys.version_info.major == 2:
+            from urllib import urlretrieve
+        elif sys.version_info.major == 3:
+            from urllib.request import urlretrieve
+        urlretrieve(url, 'cifar10.zip')
+        import zipfile
```

Comment:
confirmed. It works on Python 2.7.10 and 3.5.1

On Sat, Jun 4, 2016 at 2:53 AM, CharlesMei <notifications@github.com> wrote:

> In example/image-classification/train_cifar10.py
> <https://github.com/dmlc/mxnet/pull/2325#discussion_r65758312>:
>
> >      os.chdir(data_dir)
> >      if (not os.path.exists('train.rec')) or \
> >         (not os.path.exists('test.rec')) :
> > -        os.system("wget http://webdocs.cs.ualberta.ca/~bx3/data/cifar10.zip")
> > -        os.system("unzip -u cifar10.zip")
> > -        os.system("mv cifar/* .; rm -rf cifar; rm cifar10.zip")
> > +        url = 'http://webdocs.cs.ualberta.ca/~bx3/data/cifar10.zip'
> > +        if sys.version_info.major == 2:
> > +            from urllib import urlretrieve
> > +        elif sys.version_info.major == 3:
> > +            from urllib.request import urlretrieve
> > +        urlretrieve(url, 'cifar10.zip')
> > +        import zipfile
>
> Yes, zipfile is in python standard library.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dmlc/mxnet/pull/2325/files/ce5b4b9f8f789db8cb6799de341726d60ce3caa1#r65758312>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABFI4WfsBu9cFwBt8TuOMteeMFXLep6wks5qIHhBgaJpZM4ItTgY>
> .
>



-- 
HONG Chuntao
System Research Group
Microsoft Research Asia